src/ebmc/neural_liveness.cpp

```diff
@@ -0,0 +1,234 @@
+/*******************************************************************\
+
+Module: Neural Liveness
+
+Author: Daniel Kroening, dkr@amazon.com
+
+\*******************************************************************/
+
+#include "neural_liveness.h"
+
+#include <util/prefix.h>
+#include <util/run.h>
+#include <util/string2int.h>
+#include <util/tempdir.h>
+#include <util/tempfile.h>
+
+#include "ebmc_error.h"
+#include "random_traces.h"
+#include "ranking_function.h"
+#include "report_results.h"
+
+#include <fstream>
+
+/*******************************************************************\
+
+   Class: neural_livenesst
+
+ Purpose:
+
+\*******************************************************************/
+
+class neural_livenesst
+{
+public:
+  neural_livenesst(const cmdlinet &_cmdline, message_handlert &_message_handler)
+    : cmdline(_cmdline), message(_message_handler)
+  {
+  }
+
+  int operator()();
+
+protected:
+  const cmdlinet &cmdline;
+  messaget message;
+  transition_systemt transition_system;
+  ebmc_propertiest properties;
+
+  void validate_properties();
+  void sample(temp_dirt &);
+  exprt guess(ebmc_propertiest::propertyt &, const temp_dirt &);
+  tvt verify(ebmc_propertiest::propertyt &, const exprt &candidate);
+};
+
+int neural_livenesst::operator()()
+{
+  if(!cmdline.isset("neural-engine"))
+    throw ebmc_errort() << "give a neural engine";
+
+  int result = get_transition_system(
+    cmdline, message.get_message_handler(), transition_system);
+  if(result != -1)
+    return result;
```

Comment:
Perhaps use exceptions, as is now done everywhere else?

Reply:
Yes, coming

Reply:
#140